mods/pbj_pup/init.lua

```diff
@@ -0,0 +1,131 @@
+
+--[[
+
+  Minetest's official Peanut Butter & Jelly Pup mod
+
+]]--
+
+if minetest.setting_getbool("pbj_pup_disable") then
+	return
+end
+
+local function howl(ttl, player)
+	if not player then
+		return
+	end
+	ttl = ttl - 15
+	if ttl < 0 then
+		return
+	end
+
+	minetest.sound_play("pbj_pup_howl", {object = player, loop = false})
+	minetest.do_item_eat(5, nil, ItemStack("pbj_pup:pbj_pup"), player, nil)
+
+	minetest.after(15, howl, ttl, player)
+end
+
+--
+-- nodes
+--
+minetest.register_node("pbj_pup:pbj_pup", {
+	description = "PB&J Pup",
+	tiles = {
+		"pbj_pup_sides.png",
+		"pbj_pup_jelly.png",
+		"pbj_pup_sides.png",
+		"pbj_pup_sides.png",
+		"pbj_pup_back.png",
+		"pbj_pup_front.png"
+	},
+	paramtype = "light",
+	light_source = default.LIGHT_MAX,
+	paramtype2 = "facedir",
+	groups = {cracky = 2},
+	is_ground_content = false,
+	legacy_facedir_simple = true,
+	sounds = default.node_sound_defaults(),
+	stack_max = 1,
+	on_use = function(itemstack, user, pointed_thing)
```

Comment:
It's the player making noises from eating these, eating by punching doesn't make sense.